aten/src/ATen/TensorIterator.cpp

```diff
@@ -978,14 +976,6 @@ void TensorIteratorBase::compute_shape(const TensorIteratorConfig& config) {
     // pick it up later in the operands.
     if (config.resize_outputs_ && op.is_output) continue;
     auto shape = op.tensor.sizes();
-    if (shape.size() == 0) {
-      has_scalars = true;
-    } else {
-      has_tensors = true;
-    }
-    if (has_scalars && has_tensors) {
-      all_ops_same_shape_ = false;
-    }
     if (shape_.empty()) {
```

Comment:
ah, the test failed because this would still returns true if the current shape is scalar. So if it's `wrapped_scalar + tensor` , it will not set `all_ops_same_shape_`.

Specialize this method based on arity (unary and binary) seems to be the right approach.